tests/src/analysis/testqgsprocessingalgs.cpp

```diff
@@ -7074,6 +7076,47 @@ void TestQgsProcessingAlgs::exportMeshTimeSeries()
   outputFile.close();
 }
 
+void TestQgsProcessingAlgs::splitVectorLayer()
+{
+  QgsVectorLayer *layer = new QgsVectorLayer( QStringLiteral( "Point?crs=epsg:4326&field=pk:int&field=col1:string" ), QStringLiteral( "vl" ), QStringLiteral( "memory" ) );
+  QVERIFY( layer->isValid() );
+
+  QgsFeature f;
+  f.setAttributes( QgsAttributes() << 1 << QVariant() );
+  f.setGeometry( QgsGeometry::fromWkt( QStringLiteral( "Point (0 0)" ) ) );
+  layer->dataProvider()->addFeature( f );
+  f.setAttributes( QgsAttributes() << 2 << QStringLiteral( "" ) );
```

Comment:
```suggestion
  f.setAttributes( QgsAttributes() << 2 << QString( "" ) );
```

QStringLiteral( "" ) causes warnings on MSVC builds

Reply:
Fixed